pkg/apis/eventing/v1/trigger_validation.go

```diff
@@ -43,22 +48,49 @@ type RabbitTrigger struct {
 	Client versioned.Interface
 }
 
-var (
-	_ apis.Validatable = (*RabbitTrigger)(nil)
-)
+var _ apis.Validatable = (*RabbitTrigger)(nil)
 
 func (t *RabbitTrigger) Validate(ctx context.Context) *apis.FieldError {
 	broker, err := t.Client.EventingV1().Brokers(t.Namespace).Get(ctx, t.Spec.Broker, metav1.GetOptions{})
 	if err != nil {
 		return nil
 	}
+
 	bc, ok := broker.GetAnnotations()[eventingv1.BrokerClassAnnotationKey]
 	if !ok || bc != BrokerClass {
 		// Not my broker
 		return nil
 	}
+
+	// if prefetch is set, validate it
+	// if it isn't then the default value (1) is used
+	prefetch, ok := t.GetAnnotations()[prefetchAnnotation]
+	if ok {
+		prefetchInt, err := strconv.Atoi(prefetch)
+		if err != nil {
+			return &apis.FieldError{
+				Message: "Failed to parse valid int from prefetchAnnotation",
+				Paths:   []string{"metadata", "annotations", prefetchAnnotation},
+				Details: err.Error(),
+			}
+		}
+
+		if prefetchInt < 1 || prefetchInt > 1000 {
```

Comment:
Why is > 1000 invalid?

Reply:
Based on RabbitMQ benchmarking done by @Vanlightly in 2020 - https://blog.rabbitmq.com/posts/2020/05/quorum-queues-and-flow-control-single-queue-benchmarks - _there is not much benefit to a prefetch > 1000_ . This applies to any queue type. Thanks @lukebakken!

https://github.com/knative-sandbox/eventing-rabbitmq/pull/522#issuecomment-981852719 & https://github.com/knative-sandbox/eventing-rabbitmq/pull/522#issuecomment-981816474